### PR TITLE
8284908: Refine diagnostic positions for DCErroneous

### PR DESCRIPTION
**Pull request by pavelrappo**
Before:
```
Standard Doclet version 18+36-2087
Building tree for all the packages and classes...
Generating /tmp/whatever/ExampleDiagnostics.html...
ExampleDiagnostics.java:5: error: unexpected content
     * {@docRoot a}
       ^
ExampleDiagnostics.java:11: error: unexpected content
     * {@inheritDoc a}
       ^
ExampleDiagnostics.java:17: error: '>' expected
     * @param <T param
                     ^
```
After:
```
Standard Doclet version 19-internal-2022-04-07-1105474.pavelrappo...
Building tree for all the packages and classes...
Generating /tmp/whatever/ExampleDiagnostics.html...
ExampleDiagnostics.java:5: error: unexpected content
     * {@docRoot a}
                 ^
ExampleDiagnostics.java:11: error: unexpected content
     * {@inheritDoc a}
                    ^
ExampleDiagnostics.java:17: error: '>' expected
     * @param <T param
                ^
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8284908](https://bugs.openjdk.java.net/browse/JDK-8284908): Refine diagnostic positions for DCErroneous

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to 9bcb0a22d33d914da4e82e4e996e15c9a7509024

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8264/head:pull/8264` \
`$ git checkout pull/8264`

Update a local copy of the PR: \
`$ git checkout pull/8264` \
`$ git pull https://git.openjdk.java.net/jdk pull/8264/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8264`

View PR using the GUI difftool: \
`$ git pr show -t 8264`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8264.diff">https://git.openjdk.java.net/jdk/pull/8264.diff</a>

</details>

### CODE REVIEW

**Comment by pavelrappo** [comment]
A drive-by observation: shouldn't we allow whitespace in between the type variable and the closing `>`?

Currently, this is valid:
```
@param < T> param
```
while this is not:
```
@param <T > param
```

If we decide to allow whitespace in between type variable and the closing `>`, it should be done in a separate PR.

File: test/langtools/tools/javac/doctree/DocCommentTester.java [hunk 1]
```diff
@@ -727,7 +727,9 @@ public Void visitOther(DocTree node, Void p) {
              */
             void header(DocTree node) {
                 indent();
-                out.println(simpleClassName(node) + "[" + node.getKind() + ", pos:" + ((DCTree) node).pos);
+                var n = (DCTree) node;
+                out.println(simpleClassName(node) + "[" + node.getKind() + ", pos:" + n.pos +
+                        (n.getPreferredPosition() > n.pos ? ", prefPos:" + n.getPreferredPosition() : ""));
```
Comment (jonathan-gibbons) [inline test/langtools/tools/javac/doctree/DocCommentTester.java:732]: Being paranoid, I suggest using `!=` instead of `>`

**Review by jonathan-gibbons — APPROVED** [review]
Looks generally good for me. One small suggestion in the base `DocCommentTester` class

File: test/langtools/tools/javac/doctree/DocCommentTester.java (hunk 1 above)
Comment (pavelrappo) [reply test/langtools/tools/javac/doctree/DocCommentTester.java]: If this suggestion of yours passes tests, I will integrate it.

**Pull request closed without merging**